ui/app/components/alert-banner.js

```diff
@@ -1,13 +1,27 @@
 import Component from '@ember/component';
 import { computed } from '@ember/object';
-
 import { messageTypes } from 'vault/helpers/message-types';
 
+/**
+ * @module AlertBanner
+ * `AlertBanner` components are used to inform users of important messages.
+ *
+ * @example
+ * ```js
+ * <AlertBanner @type="danger" @message="{{model.keyId}} is not a valid lease ID"/>
+ * ```
+ *
+ * @property [AlertBanner.type=null] {String} - The banner type. This comes from the message-types helper.
+ * @property message=null {String} - The message to display within the banner.
+ *
+ * @see {@link https://github.com/hashicorp/vault/search?l=Handlebars&q=AlertBanner|Uses of AlertBanner}
```

Comment:
Can we generate these links and append or prepend them to the MD instead of extracting them here? Seems like the sort of thing that would get missed / left off / easy to get wrong.

Reply:
that is an excellent idea. madalyn is going to add this to the generator. thank you for the suggestion!